### ISSUE TITLE
resource-query-function slot is not used if buffer have a blocker-mode enabled

### ISSUE THREAD

**Issue by chrisboeg** [atlas-engineer/next#518]
If I add the following to my `init.lisp`:

```
(defclass my-buffer (buffer)
  ((default-modes :initform (list get-default 'buffer 'default-modes))
   (resource-query-function
     :initform (lambda (&rest args)
                 (progn
                   (log:info "args: ~a" args)
                   nil)))))

(setf *buffer-class* 'my-buffer)
```

and then run next in a shell, I can see the log-output from my `resource-query-function` in the `stdout` from next and indeed no urls are allowed to load. However if I add a `blocker-mode` like so:

```
(define-mode my-blocker-mode (next/blocker-mode:blocker-mode)
  "A docstring."
  ((hostlists :initform (list next/blocker-mode:*default-host-list*))))

(defclass my-buffer (buffer)
  ((default-modes :initform (append
                              '(my-blocker-mode)
                              (get-default 'buffer 'default-modes)))
   (resource-query-function
     :initform (lambda (&rest args)
                 (progn
                   (log:info "args: ~a" args)
                   nil)))))

(setf *buffer-class* 'my-buffer)
```

now my `resource-query-function` is completely ignored. It seems like the `blocker-mode` overwrites the `resource-query-function` slot behind the scene, which I think seems like a bug.

**Information**
- Next version (from =M-x next-version= or =next --version=): 1.4.0
- WebKitGTK+ / PyQt-Webengine version: WebKitGTK+

**Comment by Ambrevar** [comment]
Hi Chris,

yes, blocker-mode sets resource-query-function.

If you want to use both blocker-mode and your own query function, the
solution is simple: define your custom my-blocker-mode which inherits
from blocker-mode and assign it your own resource-query-function.  This
function will probably fall back on blocker-mode's query function.

As you realized first hand, this is not the greatest design decision as
it can trip users easily.

A while back I though we should make resource-query-function a list of
functions and run them all in order.
At this point, it looks very much like a hook, and now that we've merged
the overpowered hook system, maybe we could leverage that.

**Comment by chrisboeg** [comment]
Thank you for the tip on how to work around it.

I at least think it would be a good idea to allow multiple functions in some way, because as it is now is confusing until you read the code.

**Comment by Ambrevar** [comment]
Maybe reopen this issue until we've implemented a better approach, for
instance with hooks.

**Comment by chrisboeg** [comment]
Sorry.

**Comment by Ambrevar** [comment]
Fixed with 3d383445cab3da5f7c2c5208a83c02c825e88973.
You can now add as many handlers as you want to `request-resource-hook`.
See blocker-mode for an example.
